src/Fixer/Phpdoc/PhpdocTrimAfterDescriptionFixer.php

```diff
@@ -0,0 +1,159 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_DOC_COMMENT);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            $doc = new DocBlock($token->getContent());
+            $summaryEnd = (new ShortDescription($doc))->getEnd();
+
+            if (null === $summaryEnd) {
+                continue;
+            }
+
+            $this->fixSummary($doc, $summaryEnd);
+            $this->fixDescription($doc, $summaryEnd);
+
+            $tokens[$index] = new Token([T_DOC_COMMENT, $doc->getContent()]);
+        }
+    }
+
+    private function fixSummary(DocBlock $doc, $summaryEnd)
+    {
+        $nonBlankLineAfterSummary = $this->findNonBlankLine($doc, $summaryEnd);
+
+        $this->removeExtraBlankLinesBetween($doc, $summaryEnd, $nonBlankLineAfterSummary);
+    }
+
+    private function fixDescription(DocBlock $doc, $summaryEnd)
+    {
+        $annotationStart = $this->findFirstAnnotation($doc);
+        // assuming the end of the Description appears before the first Annotation
+        $descriptionEnd = $this->reverseFindLastUsefulContent($doc, $annotationStart);
+
+        if (null === $descriptionEnd || $summaryEnd === $descriptionEnd) {
+            return; // no Description
+        }
+
+        $this->removeExtraBlankLinesBetween($doc, $descriptionEnd, $annotationStart);
+    }
+
+    private function removeExtraBlankLinesBetween(DocBlock $doc, $from, $to)
+    {
+        for ($index = $from + 1; $index < $to; ++$index) {
+            $line = $doc->getLine($index);
+            $next = $doc->getLine($index + 1);
+            $this->removeExtraBlankLine($line, $next);
+        }
+    }
+
+    private function removeExtraBlankLine(Line $current, Line $next)
+    {
+        if (!$current->isTheEnd() && !$current->containsUsefulContent()
+            && !$next->isTheEnd() && !$next->containsUsefulContent()) {
+            $current->remove();
+        }
+    }
+
+    private function findNonBlankLine(DocBlock $doc, $after)
+    {
+        foreach ($doc->getLines() as $index => $line) {
+            if ($index <= $after) {
+                continue;
+            }
+
+            if ($line->containsATag() || $line->containsUsefulContent() || $line->isTheEnd()) {
+                return $index;
+            }
+        }
+    }
+
+    private function findFirstAnnotation(DocBlock $doc)
+    {
+        $index = null;
+        foreach ($doc->getLines() as $index => $line) {
+            if ($line->containsATag()) {
+                return $index;
+            }
+        }
+
+        return $index; // no Annotation, return the last line
+    }
+
+    private function reverseFindLastUsefulContent(DocBlock $doc, $from)
```

Comment:
to help understandability of the code, could you add docblocks describing the type of parameters on methods that aren't fully-typed like this one?